java/code/src/com/redhat/rhn/common/security/SessionSwap.java

```diff
@@ -147,7 +147,7 @@ public static String rhnHmacData(List<String> text) {
 
         String joinedText = StringUtils.join(text.iterator(), "\0");
 
-        String retval = HMAC.sha1(joinedText, swapKey.toString());
+        String retval = HMAC.sha256(joinedText, swapKey.toString());
```

Comment:
I did not understand your explanation about restoring https://github.com/uyuni-project/uyuni/pull/7276/files is not needed. That was on master and now it's not there anymore :-?

Don't you need to add
```
        if (log.isDebugEnabled()) {
            log.debug("Data     : [{}]", joinedText);
            log.debug("Key      : [{}]", swapKey);
        }
```
On top of this again?

Reply:
it's why I asked for a second check from you and Michael... 
I looked at the git log in https://github.com/uyuni-project/uyuni/commits/fe894dfc391f95e30ebe9fd5b489891c143176c5/java/code/src/com/redhat/rhn/common/security/SessionSwap.java

in 2e8919451946803b61481f0de51be7058ca8d101 Cedric applied the removal and he reverted it in ce0874791849989b6dff7f9ef7fb945c4922ef3a for having my backport. but if I revert my #7276 I'll reintroduce something that is not supposed to be there anymore because that is part of bsc#1210086

Reply:
> All I know is that it was in master and you removed it :-)

yes, I  know, but looking at the status of master before  my changes, I really think that I don't need to add back #7276